apps/antalmanac/src/lib/download.ts

```diff
@@ -0,0 +1,318 @@
+    'RRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=2SU\n' +
+    'END:DAYLIGHT\n' +
+    'BEGIN:STANDARD\n' +
+    'TZOFFSETFROM:-0700\n' +
+    'TZOFFSETTO:-0800\n' +
+    'TZNAME:PST\n' +
+    'DTSTART:19701101T020000\n' +
+    'RRULE:FREQ=YEARLY;BYMONTH=11;BYDAY=1SU\n' +
+    'END:STANDARD\n' +
+    'END:VTIMEZONE\n' +
+    'BEGIN:VEVENT';
+
+/**
+ * @example [YEAR, MONTH, DAY, HOUR, MINUTE]
+ */
+export type DateTimeArray = [number, number, number, number, number];
+
+/**
+ * @example [YEAR, MONTH, DAY]
+ */
+export type YearMonthDay = [number, number, number];
+
+/**
+ * Get the days that a class occurs.
+ * Given a string of days, convert it to a list of days in ics format
+ *
+ * @example ("TuThF") -> ["TU", "TH", "FR"]
+ */
+export function getByDays(days: string): string[] {
+    return daysOfWeek.filter((day) => days.includes(day)).map((day) => translateDaysForIcs[day]);
+}
+
+/**
+ * Get the start date of a class
+ * Given the term and bydays, this computes the start date of the class.
+ *
+ * @example ("2021 Spring", 'Tu') -> [2021, 3, 30]
+ */
+export function getClassStartDate(term: string, bydays: string[]) {
+    // Get the start date of the quarter (Monday)
+    const quarterStartDate = new Date(...quarterStartDates[term]);
+
+    // The number of days since the start of the quarter.
+    let dayOffset: number;
+
+    // Since Fall quarter starts on a Thursday,
+    // the first byday and offset will be different from other quarters.
+    // Sort by this ordering: [TH, FR, SA, SU, MO, TU, WE]
+    if (getQuarter(term) === 'Fall') {
+        bydays.sort((day1, day2) => {
+            return fallDaysOffset[day1] - fallDaysOffset[day2];
+        });
+        dayOffset = fallDaysOffset[bydays[0]];
+    } else {
+        dayOffset = daysOffset[bydays[0]];
+    }
+
+    // Add the dayOffset to the quarterStartDate
+    // Date object will handle potential overflow into the next month
+    quarterStartDate.setDate(quarterStartDate.getDate() + dayOffset);
+
+    // Return [Year, Month, Date]
+    return dateToIcs(quarterStartDate);
+}
+
+/**
+ * Convert a Date object to ics format, i.e. [YYYY, MM, DD]
+ */
+export function dateToIcs(date: Date) {
+    return [
+        date.getFullYear(),
+        date.getMonth() + 1, // Add 1 month since it is 0-indexed
+        date.getDate(),
+    ] as YearMonthDay;
+}
+
+/**
+ * Get the start and end datetime of the first class.
+ *
+ * @example ([2021, 3, 30], " 4:00-4:50p") -> [[2021, 3, 30, 16, 0], [2021, 3, 30, 16, 50]]
+ */
+export function getFirstClass(
+    date: YearMonthDay,
+    startTime: HourMinute,
+    endTime: HourMinute
+): [DateTimeArray, DateTimeArray] {
+    const [classStartTime, classEndTime] = parseTimes(startTime, endTime);
+    return [
+        [...date, ...classStartTime],
+        [...date, ...classEndTime],
+    ];
+}
+
+/**
+ * Get the start and end datetime of an exam
+ *
+ * @example ("Mon Jun 7 10:30-12:30pm", "2019") -> [[2019, 6, 7, 10, 30], [2019, 6, 7, 12, 30]]
```

Comment:
I believe this example should be WebsocFinalExam (the type) instead of the string

Reply:
Ok, I made the example more comprehensive.